File: src/test/functional/org/apache/prepbuddy/functionaltests/framework/TestReport.java
package org.apache.prepbuddy.functionaltests.framework;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.TreeSet;

public class TestReport implements Serializable {

    private List<DatasetTestResults> results = new ArrayList<>();

    public void add(DatasetTestResults testResults) {
        results.add(testResults);
    }

    public void show() {
        for (DatasetTestResults result : results) {
            System.out.println("result = " + result.getOutcome());
        }
    }
}

File: src/test/functionaltest/org/apache/datacommons/prepbuddy/functional/tests/framework/TestReport.scala
package org.apache.datacommons.prepbuddy.functional.tests.framework

import scala.collection.mutable
import scala.collection.mutable.ListBuffer

class TestReport {
    private val results: ListBuffer[TestResult] = mutable.ListBuffer.empty

    def add(testResults: TestResult): Unit = results += testResults

    def show(): Unit = {
        println("=====================================================================================================")
        results.foreach(result => println(result.getOutcome + "\r\n"))
        val failedCount: Int = results.count(_.isFailed)
        val passedCount: Int = results.size - failedCount
        println("-> " + passedCount + " Test Passed and " + failedCount + " Test has been Failed.")
        println("=====================================================================================================")
    }

    def hasFailingTest: Boolean = results.exists(_.isFailed)
}
